### ISSUE TITLE
Layout/ClassStructure cop should allow grouping macros by their visibility

### ISSUE THREAD

**Issue by gPrado** [rubocop-hq/rubocop#5465]
## Expected behavior

There should be a way to configure different visibility levels for macros to avoid false positives with the ordering of protected/private macros.

## Actual behavior

Currently, there is no way to make such configuration.

## Steps to reproduce the problem

Given the following configuration:

```yaml
Layout/ClassStructure:
  Categories:
    - attribute:
      - attr_accessor
      - attr_writer
      - attr_reader
  ExpectedOrder:
    - attribute
    - initializer
```
And the following class definition:

```ruby
class Person
  attr_accessor :name

  def initialize(name)
    @name = name
  end

  private

  attr_accessor :age
end
```

Rubocop gives the following output:

```
person.rb:10:3: C: Layout/ClassStructure: attribute is supposed to appear before initializer.
  attr_accessor :age
  ^^^^^^^^^^^^^^^^^^
```

## RuboCop version

```
$ rubocop -V
0.52.1 (using Parser 2.4.0.2, running on ruby 2.3.0 x86_64-darwin16)
```
